website/docs/proof-rules/ast.md

```diff
@@ -4,4 +4,209 @@ sidebar_position: 5
 
 # Almost-Sure Termination
 
-TODO.
\ No newline at end of file
+_Almost-sure termination_ (AST for short) means that a program terminates with probability one.
+In our probabilistic setting, this does not necessarily mean that all executions terminate (we would call that _certain termination_), but only that the expected value to reach a terminating state is one.
+In terms of weakest pre-expectations, this means that `wp[C](1) = 1` holds for a program `C`.
+For a nice overview of details and proof rules that are available in the literature, we refer to [Chapter 6 of Benjamin Kaminski's PhD thesis](https://publications.rwth-aachen.de/record/755408/files/755408.pdf#page=139).[^1]
+
+:::info
+
+In this documentation, we'll refer to almost-sure termination as probability of termination _from all initial states_.
+This is often called _universal almost-sure termination_.
+However, all definitions and proof rules can be refined to a subset of all possible initial states.
+
+:::
+
+In Caesar, there are several ways to prove almost-sure termination:
+
+```mdx-code-block
+import TOCInline from '@theme/TOCInline';
+
+<TOCInline toc={toc} maxHeadingLevel="2" />
+```
+
+## Lower Bounds on Weakest Pre-Expectations
+
+With an encoding of standard weakest pre-expectation reasoning (`wp`), it is sufficient to encode the proof of `1 <= wp[C](1)` in HeyVL, i.e. verify a `proc` with `pre` and `post` of value `1`.
+Since `wp[C](1) <= 1` always holds (by a property often called _feasibility_), `1 <= wp[C](1)` implies `wp[C](1) = 1`, i.e. almost-sure termination.
+
+Lower bounds on weakest pre-expectations need a proof rule like [omega-invariants](./omega-invariants.md) to reason about loops.
+To _refute_ a lower bound on weakest pre-expectations, [unrolling](./unrolling.md) (also known as bounded model checking) can be used.
+
+## A New Proof Rule for Almost-Sure Termination (`@ast` Annotation)
+
+Caesar supports the _"new proof rule for almost-sure termination"_ by McIver et al. as a built-in encoding.
+You can find the [extended version of the paper on arxiv](https://arxiv.org/pdf/1711.03588.pdf).
+The paper was [published at POPL 2018](https://dl.acm.org/doi/10.1145/3158121).
+
+The proof rule is based on a real-valued _loop variant_ (also known as _super-martingale_) that decreases randomly with a certain probability in each iteration by a certain amount.
```

Comment:
Suggestion: include the names of mathematical objects in the text, i.e. "loop variant $V$ that decreases randomly with probability $prob(v)$ in each iteration by a certain amount $decrease(v)$, where $v = V(s)$" or similar